src/graphql/data/TokenPrice.ts

```diff
@@ -1,84 +1,60 @@
 import graphql from 'babel-plugin-relay/macro'
-import { useEffect, useMemo, useState } from 'react'
-import { fetchQuery } from 'react-relay'
+import { useEffect, useMemo } from 'react'
+import { loadQuery, PreloadedQuery, usePreloadedQuery, useRelayEnvironment } from 'react-relay'
 
 import { Chain, TokenPriceQuery } from './__generated__/TokenPriceQuery.graphql'
-import environment from './RelayEnvironment'
-import { TimePeriod } from './util'
+import { TimePeriod, toHistoryDuration } from './util'
 
-const tokenPriceQuery = graphql`
-  query TokenPriceQuery($contract: ContractInput!) {
+export const tokenPriceQuery = graphql`
+  query TokenPriceQuery($contract: ContractInput!, $duration: HistoryDuration!) {
     tokens(contracts: [$contract]) {
-      market(currency: USD) {
-        priceHistory1H: priceHistory(duration: HOUR) {
-          timestamp
-          value
+      market(currency: USD) @required(action: LOG) {
+        price {
+          value @required(action: LOG)
         }
-        priceHistory1D: priceHistory(duration: DAY) {
-          timestamp
-          value
-        }
-        priceHistory1W: priceHistory(duration: WEEK) {
-          timestamp
-          value
-        }
-        priceHistory1M: priceHistory(duration: MONTH) {
-          timestamp
-          value
-        }
-        priceHistory1Y: priceHistory(duration: YEAR) {
-          timestamp
-          value
+        priceHistory(duration: $duration) {
+          timestamp @required(action: LOG)
+          value @required(action: LOG)
         }
       }
     }
   }
 `
-
-export type PricePoint = { timestamp: number; value: number }
+export type PricePoint = NonNullable<
+  NonNullable<NonNullable<NonNullable<TokenPriceQuery['response']['tokens']>[number]>['market']['priceHistory']>[number]
+>
 export type PriceDurations = Partial<Record<TimePeriod, PricePoint[]>>
 
-export function isPricePoint(p: { timestamp: number; value: number | null } | null): p is PricePoint {
-  return Boolean(p && p.value)
+export function isPricePoint(p: { timestamp: number; value: number } | null): p is PricePoint {
+  return p !== null
 }
 
-export function useTokenPriceQuery(address: string, chain: Chain): PriceDurations | undefined {
+export function useLoadTokenPriceQuery(address: string, chain: Chain, timePeriod: TimePeriod) {
+  const duration = toHistoryDuration(timePeriod)
   const contract = useMemo(() => ({ address: address.toLowerCase(), chain }), [address, chain])
-  const [prices, setPrices] = useState<PriceDurations>()
+  const queryReference = loadQuery<TokenPriceQuery>(useRelayEnvironment(), tokenPriceQuery, { contract, duration })
```

Comment:
`loadQuery` is only meant to be called on load (ie from a click event), and only meant to be called outside of the render cycle. This should instead use `useLazyLoadQuery`, and just rename `useLoadTokenPriceQuery` to `useLazyLoadTokenPriceQuery`, because the intention is to load the query once you are on the page.